### ISSUE TITLE
Bug: func-call-spacing is triggered for anonymous function declarations (Vue3 defineEmits)

### ISSUE THREAD

**Issue by scambier** [eslint/eslint#15213]
### Environment

Node version: 14.15.4
npm version: 6.14.10
Local ESLint version: 7.32.0
Global ESLint version: 7.32.0
Operating System: Win 10

### What parser are you using?

@typescript-eslint/parser

### What did you do?

In a Vue <script>, the following line:
```ts
const emit = defineEmits<{(e: 'update:modelValue', foo: Bar[]): void }>()
```
is, due to line length constraints, re-formatted into
```ts
const emit = defineEmits<{
  (e: 'update:modelValue', foo: Bar[]): void
}>()
```

The rule "func-call-spacing" is then triggered:
![image](https://user-images.githubusercontent.com/3216752/138830388-d2b09e64-64df-4848-a372-ac36c7637336.png)

```js
module.exports = {
  env: {
    browser: true,
    es2021: true
  },
  globals: {
    defineProps: 'readonly',
    defineEmits: 'readonly',
    defineExpose: 'readonly',
    withDefaults: 'readonly'
  },
  plugins: [
    'vue',
    '@typescript-eslint'
  ],
  extends: [
    'plugin:vue/base',
    'plugin:vue/vue3-essential',
    'plugin:vue/vue3-strongly-recommended',
    'standard'
  ],
  parserOptions: {
    ecmaVersion: 12,
    parser: '@typescript-eslint/parser',
    sourceType: 'module'
  },
  rules: {
    'no-unused-vars': ['warn'],
    'space-before-function-paren': ['error', {
      anonymous: 'always',
      named: 'never',
      asyncArrow: 'always'
    }],
    'vue/html-closing-bracket-newline': ['error', {
      singleline: 'never',
      multiline: 'never'
    }],
    'vue/html-self-closing': ['error', {
      html: { normal: 'never' }
    }]
  }
}

```
</details>

### What did you expect to happen?

The rule shouldn't trigger because a) the function declaration doesn't even have a name here and b) according to the [rule description](https://eslint.org/docs/rules/func-call-spacing), line returns before the parentheses are ok.

### What actually happened?

\src\components\MyComponent.vue
  80:27  error  Unexpected whitespace between function name and paren  func-call-spacing

### Participation

- [ ] I am willing to submit a pull request for this issue.
